Userland/Libraries/LibJS/Runtime/ECMAScriptFunctionObject.cpp

```diff
@@ -371,17 +381,55 @@ ThrowCompletionOr<void> ECMAScriptFunctionObject::function_declaration_instantia
             });
     }
 
-    auto arguments_object_needed = m_might_need_arguments_object;
+    // 7. Let simpleParameterList be IsSimpleParameterList of formals.
+    bool const simple_parameter_list = has_simple_parameter_list();
 
-    if (this_mode() == ThisMode::Lexical)
-        arguments_object_needed = false;
+    // NOTE: Already set above.
+    // 8. Let hasParameterExpressions be ContainsExpression of formals.
```

Comment:
Please swap, spec first NOTE second

Reply:
Thanks! And yeah, I was actually thinking about what style to use for some of them, especially the multi-line ones. Will fix, good to get clarification for future :^)